packages/ui/components/editor/plugins/ToolbarPlugin.tsx

```diff
@@ -340,23 +340,47 @@ export default function ToolbarPlugin(props: TextEditorProps) {
   };
 
   useEffect(() => {
-    editor.update(() => {
-      const parser = new DOMParser();
-      const dom = parser.parseFromString(props.getText(), "text/html");
+    if (!props.firstRender) {
+      editor.update(() => {
+        const root = $getRoot();
+        if (root) {
+          editor.update(() => {
+            const parser = new DOMParser();
+            // Create a new TextNode
+            const dom = parser.parseFromString(props.getText(), "text/html");
+
+            const nodes = $generateNodesFromDOM(editor, dom);
+            const paragraph = $createParagraphNode();
+            root.clear().append(paragraph);
+            paragraph.select();
+            $insertNodes(nodes);
+          });
+        }
+      });
+    }
+  }, [props.updateTemplate]);
 
-      const nodes = $generateNodesFromDOM(editor, dom);
+  useEffect(() => {
+    if (props.setFirstRender) {
+      props.setFirstRender(false);
```

Comment:
everything except `props.setFirstRender(false)` stayed the same in this `useEffect`